apps/freeablo/farender/renderer.cpp

```diff
@@ -233,13 +235,16 @@ namespace FARender
 
                     size_t x = position.current().first;
                     size_t y = position.current().second;
-
-                    mLevelObjects[x][y].valid = std::get<0>(state->mObjects[i])->isValid();
-                    mLevelObjects[x][y].spriteCacheIndex = std::get<0>(state->mObjects[i])->getCacheIndex();
-                    mLevelObjects[x][y].spriteFrame = std::get<1>(state->mObjects[i]);
-                    mLevelObjects[x][y].x2 = position.next().first;
-                    mLevelObjects[x][y].y2 = position.next().second;
-                    mLevelObjects[x][y].dist = position.mDist;
+                    Render::LevelObject levelObject = {
+                        std::get<0>(state->mObjects[i])->isValid(),
+                        std::get<0>(state->mObjects[i])->getCacheIndex(),
+                        std::get<1>(state->mObjects[i]),
+                        position.next().first,
+                        position.next().second,
+                        position.mDist
+                    };
+
+                    mLevelObjects[x][y].push_back(levelObject);
```

Comment:
Could maybe have some variable on actors that indicates the frame they arrived at their current position, and then sort the list by that variable?